completion_aggregator/tasks.py

```diff
@@ -59,7 +59,7 @@ def update_aggregators(username, course_key, block_keys=(), force=False):
     """
     user = User.objects.get(username=username)
     course_key = CourseKey.from_string(course_key)
-    block_keys = set(UsageKey.from_string(key) for key in block_keys)
+    block_keys = set(UsageKey.from_string(key).map_into_course(course_key) for key in block_keys)
```

Comment:
map_into_course adds the run information to the block keys, which allows them to match the data coming from the block structures.